lib/model.js

```diff
@@ -2102,7 +2102,9 @@ class Model {
         const flattenedWhere = Utils.flattenObjectDeep(options.where);
         const flattenedWhereKeys = _.map(_.keys(flattenedWhere), name => _.last(_.split(name, '.')));
         const whereFields = flattenedWhereKeys.map(name => _.get(this.rawAttributes, `${name}.field`, name));
-        const defaultFields = options.defaults && Object.keys(options.defaults).map(name => this.rawAttributes[name].field || name);
+        const defaultFields = options.defaults && Object.keys(options.defaults)
```

Comment:
Rather than hiding this, we should just throw an error if default doesnt contain valid attributes.

Reply:
I understand why that would be useful, but I think being able to pass a superset of attributes is convenient. It's also consistent with how Sequelize tends to work now, e.g. `User.foo = 1` will just ignore `foo` if it's not an attribute.

Reply:
`Model.create({unknownAttribute: ...})` is a better example. I think if Sequelize doesn't throw an error in this case, it shouldn't for `Model.findOrCreate({defaults: {unknownAttribute: ...}})`.